### ISSUE TITLE
Library update to support media

### ISSUE THREAD

**Issue by akramsarkis** [Geeks-Solutions/vue-sections#92]
- [x] We need the nuxt library to support media  
- When you click to add an image from the sections we should have the interface we have for media on the BO showing to be able
 - Search
 - Have the listing
 - Add media
 - Edit one
-  For doing the above you need to make sure  that the componant are reusable
- When I add a media or edit a media, I should have an additional button to associate the media to the section I am adding it to
- This should work as well when adding a picture from wyzywig even if you need to add an additional option to the wyzywig and hide the existing one
- When you remove a media, it should be only unlinked from the content you added it to

**Comment by mahfouzhseineweev** [comment]
> You need to update the plugin view cli to install the new library nuxt sections, so the one using nuxt module

@jfayad Should it be the same plugin that install vue-sections library ? Or we should have a separate plugin that install nuxt-sections ?

**Comment by mahfouzhseineweev** [comment]
Julien's reply: I would not multiply the libraries, instead I would have the same cli package to install the nuxt package by default and provide the ability to have the Vue one setup with an extra param.

As for the media components they are the same whether they are being used inside the vue or the next library, aren't they ?

**Comment by mahfouzhseineweev** [comment]
@jfayad The Media Meta components are relying on other components from the geeks apps bo components library like Buttons, inputs, animatedLoading, filterSelect, Table etc...  So if we want to include the Media Meta component in the nuxt sections module, we will have to move all of these components 2 which is hard to maintain when there is updates needed and it will complicate the Sections module.

**Comment by jfayad** [comment]
I think we should extract these components in a separate library that will be then used in the Geeks BO Apps library AND the nuxt-sections one (we maintain them in one place)

**Comment by mahfouzhseineweev** [comment]
@jfayad But the Geeks BO Apps library is a private library while the nuxt-sections is a public one. Also the media metacomponent is using almost all the components so why not simply using it as it is for nuxt-module ?

I would suggest to use an archived version of the Geeks BO Apps library inside the nuxt module, like this we keep our Geeks BO Apps library private and we use the archived version to build the media metacomponent.
Like this the update will still be maintained in one place and only an update will be required on the archived version

**Comment by jfayad** [comment]
what is an "archived version" this looks a lot like a branch of the library... which means you'll end up maintaining a fork and you're back to square one.

Let's move this conversation over the Bo Apps Library project to have a clear visibility of what would be needed

**Comment by akramsarkis** [comment]
When this one is ready to test, I need to test  how a private media used on the site is showing

**Comment by mahfouzhseineweev** [comment]
The issue took more time than expected because of the following:

As the new vue components library has to be working on both geeks apps bo components library and the nuxt sections module. I faced a challenge on making it work as expected on both projects without the need of making any updates on these projects. Instead now it will work out of the box when it is installed.

So the issue took more time while implementing the fix to make it work on both libraries without any requirements, plus the fact that I had to install the new library on both modules and test. Then install those libraries on Sections Website and Sections BO to test the components in action
